assets/js/ajax_search_select.js

```diff
@@ -0,0 +1,168 @@
+              $.ajax({
+                url: self.api_url + $(this).attr("value"),
+              }).done(function(data) {
+                self.addSelection(data['id'], self.parameter_list.map(
+                    item => data[item]
+                ).join(', ')
+                );
+              });
+            });
+            this.result = this.widget.find(this.settings.result_selector);
+            this.field = this.widget.find(this.settings.field_selector)
+                .on("keypress", function(event) {
+                    if (event.keyCode == 13) {
+                        event.preventDefault();
+                        self.searchOptions(true);
+                    }
+                }).on("keyup", function(event) {
+                    if (event.keyCode != 13) {
+                        clearTimeout(self.timeout);
+                        self.timeout = setTimeout(function() {
+                            self.searchOptions(true);
+                            self.field.focus();
+                        }, 500);
+                    }
+                });
+            this.search = this.widget.find(this.settings.search_selector)
+                .on("show.bs.dropdown", function(event) {
+                    self.searchOptions();
+                });
+            this.element.parents("form").on("submit", function(event) {
+                self.finish();
+            });
+            this.element.removeAttr('required');
+        },
+
+        searchOptions: function(show_dropdown) {
+            if (show_dropdown && this.result.is(":visible") === false) {
+                this.search.find("button").dropdown("toggle");
+            }
+            var selector = "option";
+            const query = this.field.val().trim();
+            const self = this;
+            if (query.length > 0) {
+              $.ajax({
+                url: this.api_url,
+                data: {
+                  "search": query,
+                },
+              }).done(function(data) {
+                const numOfEntries = data.count;
+                self.result.empty();
+                if (numOfEntries > 0) {
+                  for (let i = 0; i < numOfEntries; i++) {
+                      const result_info = data.results[i];
+                      self.result.append(
+                        $('<li>').append(
+                          $('<a>').text(
+                            self.parameter_list.map(
+                                item => result_info[item]
+                            ).join(', ')
+                          )
+                        ).click(function() {
+                          self.addSelection(
+                            result_info['id'], self.parameter_list.map(
+                                item => result_info[item]
+                            ).join(', ')
+                          )
+                        })
+                      );
+                    }
+                }
+                else {
```

Comment:
`else` should be on the same line as the ending `}` of the `if` block. Please check the indentation here since it looks suspicious.

Reply:
It seems that the Atom editor automatically uses 2 spaces indentation in the JavaScript files, while the A-plus code uses 4 spaces.

Reply:
Let's use the same indentation that the old code has used. That keeps it consistent. You may modify your Atom editor settings.